api/src/main/scala/za/co/absa/loginsvc/rest/config/auth/ServiceAccountConfig.scala

```diff
@@ -88,10 +88,14 @@ case class AwsSecretsLdapUserConfig(private val secretName: String,
 
   private def getUsernameAndPasswordFromSecret: (String, String) = {
     try {
-      val secrets = AwsSecretsUtils.fetchSecret(secretName, region, Array(usernameFieldName, passwordFieldName))
+      val secretsOption = AwsSecretsUtils.fetchSecret(secretName, region, Array(usernameFieldName, passwordFieldName))
+
+      if(secretsOption.isEmpty)
+        throw new Exception("Error retrieving username and password from from AWS Secrets Manager")
+
+      val secrets = secretsOption.get
```

Comment:
I think option processing can be done better without calling `Option.get` e.g. via

```scala
      secretsOption.fold(
        throw new Exception("Error retrieving username and password from from AWS Secrets Manager")
      ){ secrets =>
        (secrets(usernameFieldName), secrets(passwordFieldName))  
      }
```

or more obviously via
```scala
      secretsOption match {
        case None => throw new Exception("Error retrieving username and password from from AWS Secrets Manager")
        case Some(secrets) =>
          (secrets(usernameFieldName), secrets(passwordFieldName))
      }
   
```